core/trino-main/src/test/java/io/trino/sql/query/TestCorrelatedAggregation.java

```diff
@@ -460,6 +460,28 @@ public void testArrayAgg()
                 .matches("VALUES (1, null), (2, ARRAY[2])");
     }
 
+    @Test
+    public void testBoolOrAggregation()
+    {
+        // without projection
+        assertThat(assertions.query("""
+                SELECT * FROM
```

Comment:
Could we also have `TestLogicalPlanner` test correlated `exists` query similar to this? To make sure `bool_or` actually works and `maybeSkipOutputDuplicate=true`